src/backend/distributed/transaction/backend_data.c

```diff
@@ -855,6 +855,10 @@ CancelTransactionDueToDeadlock(PGPROC *proc)
  * MyBackendGotCancelledDueToDeadlock returns whether the current distributed
  * transaction was cancelled due to a deadlock. If the backend is not in a
  * distributed transaction, the function returns false.
+ * IMPORTANT: We keep some session level state to keep track of if we were
+ * cancelled because of a distributed deadlock. Calling this function also
+ * resets that state. So calling this function a second time right away will
+ * always return false.
  */
 bool
 MyBackendGotCancelledDueToDeadlock(void)
```

Comment:
maybe add `MyBackendGotCancelledDueToDeadlock(bool clearState)`

and 
```C
if (clearState)
{
MyBackendData->cancelledDueToDeadlock = false;
}
```

To prevent future edge cases as we might forgot this comment?